pkg/platform/local/client/store.go

```diff
@@ -310,7 +310,14 @@ func (s *Store) serializeAndWriteFileContents(resourcePath string, resourceConfi
 }
 
 func (s *Store) getResourcePath(resourceDir string, resourceNamespace string, resourceName string) string {
-	return path.Join(s.getResourceNamespaceDir(resourceDir, resourceNamespace), resourceName+".json")
+	resourcePath := path.Join(s.getResourceNamespaceDir(resourceDir, resourceNamespace), resourceName+".json")
+	// try to check if a function name has a space in it
+	if strings.Contains(resourcePath, " ") {
+		// we can get there only if a function was deployed with a wrong name in Nuclio 1.11.24 and earlier.
+		// if a function name had a space in it, then we created a file called <first_word> instead of the correct name
+		resourcePath = strings.Fields(resourcePath)[0]
```

Comment:
Does this mean that a possible work around is to run `nuctl delete` with the first word?
Or will we fail earlier on validating the function exists

Reply:
@TomerShor no, because we will anyway generate a file name with `json` format at the end.